youtube_dl/extractor/nexx.py

```diff
@@ -247,20 +254,26 @@ def get_cdn_shield_base(shield_type='', static=False):
         azure_manifest_url = '%s%s/%s_src%s.ism/Manifest' % (
             azure_stream_base, azure_locator, video_id, ('_manifest' if is_ml else '')) + '%s'
 
+        tokenDASH, tokenHLS = self._protection_tokens(video)
+        # TODO: is this kind of token storage still in use?
         protection_token = try_get(
             video, lambda x: x['protectiondata']['token'], compat_str)
-        if protection_token:
+        if protection_token and not (tokenHLS or tokenDASH):
             azure_manifest_url += '?hdnts=%s' % protection_token
 
+        azure_manifest_url_DASH = azure_manifest_url + '?hdnts=%s' % (tokenDASH)
+        azure_manifest_url_HLS = azure_manifest_url + '?hdnts=%s' % (tokenHLS)
```

Comment:
DRY.

Reply:
I'll create a function that adds the token to the URL.